cmsmon-tools.spec

```diff
@@ -0,0 +1,76 @@
+### RPM cms cmsmon-tools 0.3.10
+## INITENV +PATH PYTHONPATH %i/${PYTHON_LIB_SITE_PACKAGES}
+## NOCOMPILER
+
+%define pkg CMSMonitoring
+%define ver %realversion
+%define cmsmon_commands nats-sub nats-pub nats-exitcodes-termui dbs_vm
+%define flags -ldflags="-s -w -extldflags -static"
+Source0: https://github.com/dmwm/%pkg/archive/%ver.tar.gz
+
+Requires: go
+
+# RPM macros documentation
+# http://www.rpm.org/max-rpm/s1-rpm-inside-macros.html
+%prep
+%setup -D -T -b 0 -n %pkg-%ver
+
+%build
+export GOCACHE=%{_builddir}/gocache
+cd ..
+cd %pkg-%ver
+echo "build $PWD"
+ls
+mkdir -p gopath/bin
+export GOPATH=$PWD/gopath
+go get github.com/dmwm/cmsauth
+go get github.com/vkuznet/x509proxy
+go get github.com/sirupsen/logrus
+go get github.com/prometheus/client_golang/prometheus
+go get github.com/prometheus/common/log
+go get github.com/prometheus/common/version
+go get github.com/shirou/gopsutil/cpu
+go get github.com/shirou/gopsutil/mem
+go get github.com/shirou/gopsutil/load
+go get github.com/shirou/gopsutil/process
+go get github.com/go-stomp/stomp
+go get github.com/nats-io/nats.go
+go get github.com/gizak/termui/v3
+
+# build monit tools
+cd src/go/MONIT
+go build %flags monit.go
+cd -
+# build NATS tools
+cd src/go/NATS
+for cmd in %cmsmon_commands; do
+if [ -f $cmd.go ]; then
+go build %flags $cmd.go
+fi
+done
+cd -
+
+%install
+cd ../%pkg-%ver
+echo "### current dir: $PWD"
+cp src/go/MONIT/monit %i/
+for cmd in %cmsmon_commands; do
+if [ -f src/go/NATS/$cmd ]; then
+cp src/go/NATS/$cmd %i/
+fi
+done
+
+%post
+mkdir -p $RPM_INSTALL_PREFIX/cmsmon
+for cmd in %cmsmon_commands; do
+cat > $RPM_INSTALL_PREFIX/cmsmon/$cmd << EOF
```

Comment:
You just need to escape EOF here e.g. change it to `\EOF` and then you do not need to escape every `$` in the script itself.

Reply:
Again, why to create a seperate script for each if symlinks will work? Having  on script means in case we have to fix the script by have in /cvmfs then it is only one place to fix and not N scripts.

Reply:
Shahzad, you lost me in these discussions. My understanding that you want wrapper scripts because we can build on different architectures, e.g. x86 vs ppc. That's why I put wrappers back. But I said that I don't want one script/wrapper since I think it is confusing. If we should have wrappers script the script name should match actual executable and we will have one wrapper per one executable such that from a user usage point of view it will look like a command, regardless if it is run on lxplus or local laptop.

Please adjust this part is you think is appropriate based on the aforementioned arguments.

Reply:
Please see my original commit https://github.com/cms-sw/cmsdist/pull/5750/commits/d509adeb847dc813c518b6924fbaaa147b87b1ab . Yes I want wrapper but as the contents of all the wrappers are same that is why I was suggested to create one wrapper and then create symlinks points to the wrapper. At the end, for each command there will be one wrapper (it do not matter it that is actual script of symlink to script :-) ).

Do not worry I will update the spec.

Reply:
ok, now I understand. Please proceed with the change I think I've done everything I need from my side and you may finish with the single wrapper and different links.